index.html

```diff
@@ -2543,18 +2588,271 @@ <h1>
+returned from the <a>DID resolution</a> process. If no additional processing is
+required by the <a>DID URL</a>, the <a>DID document</a> is returned as the 
+<code>content</code>.
+        </p>
+        
+        <p>
+The <code>DID dereference metadata</code> is returned as a map of key-value string pairs as described in
+<a href="#metadata-structure"></a>. This component is REQUIRED and MUST NOT be empty.
+This metadata contains information about the results of
+the dereferencing process. This metadata typically changes between invocations of the <a>DID URL dereferencing</a>
+function. The keys and possible values for this metadata are defined by a registry (TBD). This
+specification defines the following keys and values:
+
+            <dl>
+                <dt>content-type</dt>
+                <dd>The mime-type of the returned content if successful.</dd>
+
+                <dt>error</dt>
+                <dd>The error code from the dereferencing process. The <a>DID resolver</a> MUST supply
+                    this value when there is an error. The possible values
+                    this field are defined by a registry (TBD). This specification defines the following
+                    error values:
+                    
+                    <dl>
+                        <dt>invalid-url</dt>
+                        <dd>The <a>DID URL</a> supplied to the <a>DID URL dereferencing</a> function does not
+                            conform to valid syntax. (See <a href="#did-url-syntax"></a>.)</dd>
+                    </dl>
+                    
+                </dd>
+            
+            </dl>
+        </p>
+        
+        <p>
+The <code>content</code> of this function's output is a byte stream of data. This byte stream MAY 
+represent a <a>DID document</a> in a conformant representation, a portion of a <a>DID document</a>
+in a conformant representation, the results of calling a service
+endpoint, or something else determined by the <a>DID method</a> and its services.
+        </p>
+        
+        <p>
+If the returned <code>content</code> is a <a>DID document</a>, the <a>DID URL dereferencer</a> MAY
+return an additional set of DID document metadata, as described in <a href="#did-resolution"></a>.
+        </p>
+        
+    </section>
+
+    <section>
+        <h2>
+Metadata structure
+        </h2>
+
+        <div class="issue" data-number="58">
+There needs to be a registry established for handling input options and rules defined for 
+how to handle an unknown input option. A similar issue applies for all metadata structures
+returned from these functions. 
+        </div>
+        
+        <p>
+Metadata structures are maps of key-value string pairs. All keys within this map
+MUST be UTF-8 unicode strings. The keys of this map MUST NOT be repeated within 
+a given metadata structure. The keys of this map MUST be compared in a 
+case-insensitive way and the case MAY be transformed. The values of this map 
```

Comment:
-1 to case-insensitive comparison. I understand why it's desired, but modern software practices often do matching based on case sensitivity... this is a strange rule. What is the reason it exists? We're worried about developers not understanding case-sensitive matching?

Reply:
This is how HTTP header values are compared. I don't see it as a necessary feature as long as a binding is still required to have a deterministic mapping between the key and the binding's representation.

Reply:
> This is how HTTP header names are compared.

Yeah, and I question whether or not we should let esoteric HTTP things leak into this spec. There are arguments for and against, but that design in HTTP comes from a time when case sensitivity didn't matter as much as it does today.

Reply:
I'm not convinced that it matters more or less today, and so I'd be fine with rolling that back if the WG wants to go that way.